src/TTImp/ProcessBuiltin.idr

```diff
@@ -129,45 +185,82 @@ checkCons c cons ty fc = case !(foldr checkCon (pure (Nothing, Nothing)) cons) o
             _ => throw $ GenericMsg fc $ "Constructor " ++ show n ++ " doesn't match any pattern for Natural."
 
 addBuiltinNat :
-    {auto c : Ref Ctxt Defs} ->
+    Ref Ctxt Defs =>
     (ty : Name) -> NatBuiltin -> Core ()
 addBuiltinNat type cons = do
-    log "builtin.Natural.addTransform" 10 $ "Add Builtin Natural transform for " ++ show type
+    log "builtin.Natural.addTransform" 10
+        $ "Add %builtin Natural transform for " ++ show type ++ "."
     update Ctxt $ record
         { builtinTransforms.natTyNames $= insert type cons
         , builtinTransforms.natZNames $= insert cons.zero MkZERO
         , builtinTransforms.natSNames $= insert cons.succ MkSUCC
         }
 
+addNatToInteger :
+    Ref Ctxt Defs =>
+    (fn : Name) ->
+    NatToInt ->
+    Core ()
+addNatToInteger fn nToI = do
+    log "builtin.NaturalToInteger.addTransforms" 10
+        $ "Add %builtin NaturalToInteger transform for " ++ show fn ++ "."
+    update Ctxt $ record
+        { builtinTransforms.natToIntegerFns $= insert fn nToI
+        }
+
 ||| Check a `%builtin Natural` pragma is correct.
 processBuiltinNatural :
-    {auto c : Ref Ctxt Defs} ->
-    {auto m : Ref MD Metadata} ->
-    {auto u : Ref UST UState} ->
+    Ref Ctxt Defs =>
     Defs -> FC -> Name -> Core ()
 processBuiltinNatural ds fc name = do
-    log "builtin.Natural" 5 $ "Processing Builtin Natural pragma for " ++ show name
+    log "builtin.Natural" 5 $ "Processing %builtin Natural " ++ show name ++ "."
     [(n, _, gdef)] <- lookupCtxtName name ds.gamma
         | [] => throw $ UndefinedName fc name
         | ns => throw $ AmbiguousName fc $ (\(n, _, _) => n) <$> ns
     let TCon _ _ _ _ _ _ dcons _ = gdef.definition
-        | def => throw $ GenericMsg fc $ "Expected a type constructor, found:\n" ++ show def
+        | def => throw $ GenericMsg fc
+            $ "Expected a type constructor, found " ++ showDefType def ++ "."
     cons <- getConsGDef ds.gamma fc dcons
-    cons <- checkCons ds.gamma cons n fc
+    cons <- checkNatCons ds.gamma cons n fc
     zero <- getFullName cons.zero
     succ <- getFullName cons.succ
-    n <- getFullName name
     addBuiltinNat n $ MkNatBuiltin {zero, succ}
 
+||| Check a `%builtin NaturalToInteger` pragma is correct.
+processNatToInteger :
+    Ref Ctxt Defs =>
+    Defs -> FC -> Name -> Core ()
+processNatToInteger ds fc fn = do
+    log "builtin.NaturalToInteger" 5 $ "Processing %builtin NaturalToInteger " ++ show fn ++ "."
+    [(n, _, gdef)] <- lookupCtxtName fn ds.gamma
+        | [] => throw $ UndefinedName fc fn
```

Comment:
Here you can use `undefinedName fc fn` instead of `throw $ UndefinedName fc fn` and Idris
will try to suggest names that look similar in case the user made a typo.